guava/src/main/java/com/fasterxml/jackson/datatype/guava/deser/RangeDeserializer.java

```diff
@@ -188,9 +216,50 @@ public Range<?> deserialize(JsonParser p, DeserializationContext context)
             }
             return RangeFactory.all();
         } catch (IllegalStateException e) {
-            context.reportBadDefinition(handledType(), e.getMessage());
+            return context.reportBadDefinition(getValueType(context), e.getMessage());
+        }
+    }
+
+    private Range<?> deserializeRangeFromString(DeserializationContext context, JsonParser p)
+            throws IOException
+    {
+        String rangeInterval = p.getText();
+
+        if (rangeInterval.isEmpty()) {
             return null;
         }
+
+        if (isValidBracketNotation(rangeInterval)) {
+            BoundType lowerBoundType = rangeInterval.startsWith("[") ? BoundType.CLOSED : BoundType.OPEN;
+            BoundType upperBoundType = rangeInterval.endsWith("]") ? BoundType.CLOSED : BoundType.OPEN;
+
+            rangeInterval = rangeInterval.substring(1, rangeInterval.length() - 1);
+            String[] parts = PATTERN_DOUBLE_DOT.split(rangeInterval);
+
+            if (parts.length == 2) {
+                boolean isLowerInfinite = parts[0].equals("-∞");
+                boolean isUpperInfinite = parts[1].equals("+∞");
+
+                if (isLowerInfinite && isUpperInfinite) {
+                    return RangeFactory.all();
+                } else if (isLowerInfinite) {
+                    return RangeFactory.upTo(Integer.parseInt(parts[1]), upperBoundType);
+                } else if (isUpperInfinite) {
+                    return RangeFactory.downTo(Integer.parseInt(parts[0]), lowerBoundType);
+                } else {
+                    return RangeFactory.range(Integer.parseInt(parts[0]), lowerBoundType,
+                            Integer.parseInt(parts[1]), upperBoundType);
+                }
+            }
+        } else {
```

Comment:
Like you suggested, We got two error cases now. One pops up for wrong bracket ranges like `[4..5`. 

The other, on line 256, shows up mostly for wrong delimiters like `[4, 6]` or just generic invalid representation.